### PR TITLE
Detox/E2E: Fix android build command and update deps

### PR DESCRIPTION
**Pull request by josephbaylon**
#### Summary
- Applied patches for `react-native-mmkv-storage` and `rudder-sdk-react-native`
- Updated detox dependencies
 
#### Ticket Link
NA

#### Screenshots
Screen Shot 2021-12-15 at 5 52 35 PM

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by saturninoabril** [comment]
Thanks @josephbaylon! I got a successful build but can't run Detox test. Did you try running Detox on fresh emulator? I don't think `assembleAndroidTest` could be removed but I maybe be totally wrong.

**Review by enahum — APPROVED** [review]

**Comment by josephbaylon** [comment]
@saturninoabril Yeah, I did delete my old emulator and created a new one using `npm run e2e:android-create-emulator`. Did your run `npm run start` on the mattermost-mobile terminal before running `npm run e2e:android-test`? I just tried it again and it works for me on a clean slate.

**Comment by saturninoabril** [comment]
Yes, I have it running. Still not working on me as of this writing. What I encountered was that the app and test app (`app-debug-androidTest.apk`) failed to install in the emulator. See error logs:
Screen Shot 2021-12-16 at 6 26 30 PM

**Comment by josephbaylon** [comment]
@saturninoabril I now see the issue. I put back the `assembleAndroidTest  -DtestBuildType=debug` locally but getting the original error below,

```
> Task :react-native-mmkv-storage:mergeDebugAndroidTestJavaResource FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-mmkv-storage:mergeDebugAndroidTestJavaResource'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.MergeJavaResWorkAction
   > 24 files found with path 'META-INF/MANIFEST.MF' from inputs:
      - /Users/josephbaylon/.gradle/caches/transforms-3/1dc2278d2fd0ac13be09fd06b952fb20/transformed/jetified-infer-annotation-0.18.0.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/240303040620a233ae5f69095dd804b1/transformed/jetified-jsr305-3.0.2.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/651bc1db3992a0d759451dc81fa2c10c/transformed/jetified-kotlin-annotations-jvm-1.3.72.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/e0b8fc3f1e11af1ba413f505530923d0/transformed/jetified-proguard-annotations-1.19.0.jar
      - /Users/josephbaylon/.gradle/caches/modules-2/files-2.1/androidx.collection/collection/1.0.0/42858b26cafdaa69b6149f45dfc2894007bc2c7a/collection-1.0.0.jar
      - /Users/josephbaylon/.gradle/caches/modules-2/files-2.1/androidx.arch.core/core-common/2.0.0/bb21b9a11761451b51624ac428d1f1bb5deeac38/core-common-2.0.0.jar
      - /Users/josephbaylon/.gradle/caches/modules-2/files-2.1/androidx.lifecycle/lifecycle-common/2.0.0/e070ffae07452331bc5684734fce6831d531785c/lifecycle-common-2.0.0.jar
      - /Users/josephbaylon/.gradle/caches/modules-2/files-2.1/androidx.annotation/annotation/1.1.0/e3a6fb2f40e3a3842e6b7472628ba4ce416ea4c8/annotation-1.1.0.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/20172e744f033f29c6d28d089be29ed6/transformed/jetified-soloader-0.10.1/jars/classes.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/58099db10b259ec36ec45f4f70571591/transformed/jetified-annotation-0.10.1.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/ec967da3a71eaa758d1929dde839267c/transformed/jetified-fbjni-java-only-0.2.2.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/31f7527b27c09d7f74bca0e9a98cd956/transformed/jetified-nativeloader-0.10.1.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/0f2bc3a5aba3dfa5197b8e7acad93405/transformed/jetified-bolts-tasks-1.4.0.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/43cbb2b0aed4fb6f529acdf40d494aec/transformed/jetified-okhttp-urlconnection-4.9.1.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/bc6d1f49874ac7f6ab0c796947bf2082/transformed/jetified-okhttp-4.9.1.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/791e1c739e7e232977faaeffa72a2047/transformed/jetified-okio-jvm-2.9.0.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/3526e1d9a564f906a191076ec11a7c39/transformed/jetified-kotlin-stdlib-jdk8-1.4.10.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/e45c386832b6196a4e0340751e797c01/transformed/jetified-kotlin-stdlib-jdk7-1.4.10.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/62c31116a4ed257c49ad61b9fde6e252/transformed/jetified-kotlin-stdlib-1.4.10.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/4368b4caab99b195f11fdae4c57936fb/transformed/jetified-kotlin-stdlib-common-1.4.10.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/08f3f655e45fc556cd1d3e5bc2c6b45d/transformed/jetified-annotations-13.0.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/b67885f38c7f4bde3eca4c88d5fdb5e1/transformed/jetified-secure-preferences-lib-0.1.4/jars/classes.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/428f1a419ee8f411a65025b460c4b41c/transformed/jetified-aes-crypto-0.0.4/jars/classes.jar
      - /Users/josephbaylon/.gradle/caches/transforms-3/eed2ca00989ccb3caf06b22276bc310d/transformed/jetified-gson-2.8.6.jar
     Adding a packagingOptions block may help, please refer to
     https://google.github.io/android-gradle-dsl/current/com.android.build.gradle.internal.dsl.PackagingOptions.html
     for more information
```


Can you run it with `assembleAndroidTest  -DtestBuildType=debug` and see if you get the same error? Also, if you have any idea on how to fix it, that would be cool.

**Comment by saturninoabril** [comment]
Yeah, I'm getting the same error too. Let me see if I could find something.

File: detox/.detoxrc.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "android.emu.debug": {
       "type": "android.emulator",
       "binaryPath": "../android/app/build/outputs/apk/debug/app-debug.apk",
-      "build": "cd .. && ./node_modules/.bin/jetify && cd android && ./gradlew clean assembleDebug assembleAndroidTest -DtestBuildType=debug && cd ../detox",
+      "build": "cd .. && ./node_modules/.bin/jetify && cd android && ./gradlew clean && ./gradlew assembleDebug assembleAndroidTest -DtestBuildType=debug && cd ../detox",
```
Comment (saturninoabril) [inline detox/.detoxrc.json:23]: Still include `clean` but on a separate command as suggested from an issue discussions.

File: patches/@rudderstack+rudder-sdk-react-native+1.1.0.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/node_modules/@rudderstack/rudder-sdk-react-native/android/build.gradle b/node_modules/@rudderstack/rudder-sdk-react-native/android/build.gradle
+index 1536175..1c40b7f 100644
+--- a/node_modules/@rudderstack/rudder-sdk-react-native/android/build.gradle
++++ b/node_modules/@rudderstack/rudder-sdk-react-native/android/build.gradle
+@@ -15,7 +15,7 @@ android {
+     compileSdkVersion 29
+ 
+     defaultConfig {
+-        minSdkVersion 16
++        minSdkVersion rootProject.hasProperty('minSdkVersion') ? rootProject.minSdkVersion : 19
```
Comment (saturninoabril) [inline patches/@rudderstack+rudder-sdk-react-native+1.1.0.patch:10]: Getting error on debug test build about the minSdkVersion below project requirement.

File: patches/react-native-mmkv-storage+0.6.6.patch [hunk 3]
```diff
@@ -28,3 +28,16 @@ index ffac535..4608563 100644
...
+diff --git a/node_modules/react-native-mmkv-storage/android/build.gradle b/node_modules/react-native-mmkv-storage/android/build.gradle
+index 618a7de..4aaca89 100644
+--- a/node_modules/react-native-mmkv-storage/android/build.gradle
++++ b/node_modules/react-native-mmkv-storage/android/build.gradle
+@@ -85,7 +85,7 @@ android {
+   }
+ 
+   packagingOptions {
+-    excludes = ["**/libc++_shared.so","**/libjsi.so","**/libreactnativejni.so"]
++    excludes = ["**/libc++_shared.so","**/libjsi.so","**/libreactnativejni.so", "META-INF/MANIFEST.MF"]
```
Comment (saturninoabril) [inline patches/react-native-mmkv-storage+0.6.6.patch:40]: Fix for the error encountered at https://github.com/mattermost/mattermost-mobile/pull/5879#issuecomment-995688248.
Already on package's master but not yet released - https://github.com/ammarahm-ed/react-native-mmkv-storage/commit/5dce2e9b6930cec1aa73843a78dedb4fa0545819

**Comment by saturninoabril** [comment]
@josephbaylon Please take a look if it works on you.

File: detox/.detoxrc.json (hunk 1 above)
Comment (josephbaylon) [reply detox/.detoxrc.json:23]: That make sense since individually clean and assembleDebug works

**Comment by josephbaylon** [comment]
@saturninoabril I've tried it locally and works. Thanks for your help!
@enahum Please review the patches that Saturn applied. Thanks!

**Review by enahum — APPROVED** [review]
awesome guys!

**Review by saturninoabril — APPROVED** [review]
Awesome, thanks @josephbaylon!

**Pull request merged**